fdo.go

```diff
@@ -0,0 +1,264 @@
+package desktop
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"fyne.io/fyne"
+)
+
+var iconExtensions = []string{".png", ".svg", ".xpm"}
+
+//FdoDesktop contains the information from Linux .desktop files
+type FdoDesktop struct {
+	Name     string
+	IconName string
+	IconPath string
+	Exec     string
+}
+
+//FdoResourceFormat turns a string into a usable name for fyne.Resources
+func FdoResourceFormat(name string) string {
+	str := strings.Replace(name, "-", "", -1)
+	return strings.Replace(str, "_", "", -1)
+}
+
+//FdoLookupXdgDataDirs returns a string slice of all XDG_DATA_DIRS
+func FdoLookupXdgDataDirs() []string {
+	dataLocation := os.Getenv("XDG_DATA_DIRS")
+	locationLookup := strings.Split(dataLocation, ":")
+	if len(locationLookup) == 1 && locationLookup[0] == dataLocation {
+		var fallbackLocations []string
+		fallbackLocations = append(fallbackLocations, "/usr/local/share")
+		fallbackLocations = append(fallbackLocations, "/usr/share")
+		return fallbackLocations
+	}
+	return locationLookup
+}
+
+//FdoLookupApplication looks up an application by name and returns a FdoDesktop
+func FdoLookupApplication(appName string) *FdoDesktop {
```

Comment:
keep this private too.

Reply:
There are issues with keeping this private as I mentioned above